core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1593,6 +1593,24 @@ trait ZIOFunctions extends Serializable {
    * current fiber for this operation to return non-empty lists.
    */
   final def children: UIO[IndexedSeq[Fiber[_, _]]] = descriptor.flatMap(_.children)
+
+  def fromOptionM[A](uio: UIO[Option[A]]): IO[Unit, A] = uio.flatMap {
+    case Some(a) => ZIO.succeed(a)
+    case None    => ZIO.fail(())
+  }
+
+  final def fromEitherE[E, A](io: Either[E, A]): IO[E, A] =
+    effectTotal(io).flatMap(_.fold(ZIO.fail, ZIO.succeed))
+
+  def fromEitherM[E, A](io: IO[E, Either[E, A]]): IO[E, A] = io.foldM(ZIO.fail, ZIO.fromEitherE)
+
+  def fromOptionTask[A, E](ifEmpty: => E, onError: Throwable => E)(opt: Task[Option[A]]): IO[E, A] =
```

Comment:
This feels like it's combining two operations to me. It could be written as `opt.mapError(onError).flatMap(opt => ZIO.toEither(opt.toRight(ifEmpty)))`.

I know a user requested it, but it feels too special case to be inside ZIO. I think I would omit this from the pull request.

Reply:
``` scala
 def fromOptionTask[A, E](ifEmpty: => E, onError: Throwable => E)(opt: Task[Option[A]]): IO[E, A] =
    opt.mapError(onError).flatMap(opt => ZIO.toEither(opt.toRight(ifEmpty)))
```
It doesn't compiling.
value toEither is not a member of object scalaz.zio.ZIO